### ISSUE TITLE
[Deeplabv3] Loss increase after restoring from checkpoint.

### ISSUE THREAD

**Issue by rajanieprabha** [tensorflow/models#7625]
Please go to Stack Overflow for help and support:

http://stackoverflow.com/questions/tagged/tensorflow

Also, please understand that many of the models included in this repository are experimental and research-style code. If you open a GitHub issue, here is our policy:

1. It must be a bug, a feature request, or a significant problem with documentation (for small docs fixes please send a PR instead).
2. The form below must be filled out.

**Here's why we have that policy**: TensorFlow developers respond to issues. We want to focus on work that benefits the whole community, e.g., fixing bugs and adding features. Support only helps individuals. GitHub also notifies thousands of people when issues are filed. We want them to see you communicating an interesting problem, rather than being redirected to Stack Overflow.

------------------------

### System information
- **What is the top-level directory of the model you are using**: research/deeplab
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: 
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 18.04.2 LTS
- **TensorFlow installed from (source or binary)**:Binary
- **TensorFlow version (use command below)**:1.13.1
- **Bazel version (if compiling from source)**:
- **CUDA/cuDNN version**:10.1
- **GPU model and memory**:
- **Exact command to reproduce**:

You can collect some of this information using our environment capture script:

https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh

You can obtain the TensorFlow version with

`python -c "import tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"`

### Describe the problem
When I train a model on training dataset for let's say 100,000 iterations, the loss goes down nice from 3.5 to 0.2
However, when I try to run just the eval.py using the same checkpoint on the training dataset, the restored model state is no longer the same. Loss is usually high. I haven't made any changes with saving and restoring model code.
Any leads?

### Source code / logs
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached. Try to provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Comment by manasb26** [comment]
Hi @rajanieprabha 

Have you trained the network with the same image height and width you use for evaluation? Training the network with a smaller image size than the actual size is one of the reasons for the increased loss during evaluation which I have also experienced.

**Comment by rajanieprabha** [comment]
Hey @manasb26 ,
No, that was not the problem. I couldn't reproduce this again, however, I am using smaller train crop size than the original image to train on a larger batch-size.
